src/auth/pluggable-auth-client.ts

```diff
@@ -246,72 +264,38 @@ export class PluggableAuthClient extends BaseExternalAccountClient {
       envMap.set('GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE', this.subjectTokenType);
       // Always set to 0 because interactive mode is not supported.
       envMap.set('GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE', '0');
+      if (this.outputFile) {
+        envMap.set('GOOGLE_EXTERNAL_ACCOUNT_OUTPUT_FILE', this.outputFile);
+      }
       const serviceAccountEmail = this.getServiceAccountEmail();
       if (serviceAccountEmail) {
         envMap.set(
           'GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL',
           serviceAccountEmail
         );
       }
-      executableResponse = await this.retrieveResponseFromExecutable(envMap);
+      executableResponse = await this.handler.retrieveResponseFromExecutable(
+        envMap
+      );
     }
 
-    if (executableResponse) {
-      // Check that version of response is valid.
-      if (executableResponse.version > MAXIMUM_EXECUTABLE_VERSION) {
-        throw new Error(
-          `Version of executable is not currently supported, maximum supported version is ${MAXIMUM_EXECUTABLE_VERSION}.`
-        );
-      }
-      // Check that response was successful.
-      if (!executableResponse.success) {
-        throw new ExecutableError(
-          executableResponse.errorMessage as string,
-          executableResponse.errorCode as string
-        );
-      }
-      // Check that response is not expired.
-      if (executableResponse.isExpired()) {
-        throw new Error('Executable response is expired.');
-      }
-      // Return subject token from response.
-      return executableResponse.subjectToken as string;
-    } else {
-      throw new Error('No valid response returned from executable.');
+    if (executableResponse.version > MAXIMUM_EXECUTABLE_VERSION) {
```

Comment:
why not validate executable response in the handler?

Reply:
Having the final validation here makes it so its only done once - the handler implementation here doesn't have a single method that calls both the executable and the file cache, the client calls both individually and validated after.

Reply:
I think it's better not to mix the responsibilities of the client and the handler. There's nothing stopping you from adding a method in the handler that does both (checks cache, calls executable).

Reply:
Discussed offline